packages/salesforcedx-utils-vscode/src/telemetry/reporters/appInsights.ts

```diff
@@ -94,7 +95,7 @@ export class AppInsights extends Disposable implements TelemetryReporter {
 
     this.appInsightsClient.commonProperties = this.getCommonProperties();
     if (this.uniqueUserMetrics && env) {
-      this.appInsightsClient.context.tags['ai.user.id'] = env.machineId;
+      this.appInsightsClient.context.tags['ai.user.id'] = this.userId;
```

Comment:
I think this userId would also benefit non E4D telemetry. Could this change include adding a new property to common for all to use?

If E4D requires that it use a unique property, then please add a common property and adjust that coe as well.

Reply:
FWIW I think this is the common one.  ai = appinsights not artificially intelligent :)

Reply:
This changes how we report userId in appInsights irrespective of the extensions. @peternhale Is there anything else you would want to change in how we report?

Reply:
No.